bin/improver-nbhood-land-and-sea

```diff
@@ -0,0 +1,234 @@
+        radius_or_radii = args.radius
+        lead_times = None
+    elif args.radii_by_lead_time:
+        radius_or_radii = args.radii_by_lead_time[0].split(",")
+        lead_times = args.radii_by_lead_time[1].split(",")
+
+    if args.additional_mask_filepath:
+        additional_mask = load_cube(args.additional_mask_filepath)
+        weights = load_cube(args.weights_for_collapsing_dim)
+
+        mask_cube = iris.cube.CubeList()
+        xy_slices = additional_mask.slices([cube.coord(axis='y'),
+                                            cube.coord(axis='x')])
+        for xy_slice in xy_slices:
+            mask_cube.append(xy_slice.copy(data=np.logical_and(
+                land_only.data, xy_slice.data).astype(int)))
+
+        mask_cube = merge_cubes(mask_cube)
+
+        result_land = ApplyNeighbourhoodProcessingWithAMask(
+            args.coord_for_masking, radius_or_radii, lead_times=lead_times,
+            ens_factor=args.ens_factor, sum_or_fraction=args.sum_or_fraction,
+            re_mask=False).process(cube, mask_cube)
+
+        # Reapply land-mask to mask any data spread by neighbourhooding.
+        masked_data = iris.cube.CubeList()
+        for cslice in result_land.slices(
+                [cube.coord(axis='y'), cube.coord(axis='x')]):
+            cslice.data = np.ma.masked_array(cslice.data, mask=sea_only.data)
+            masked_data.append(cslice)
+        result_land = masked_data.concatenate()
+        result_land = result_land.merge_cube()
+    else:
+        result_land = NeighbourhoodProcessing(
+            'square', radius_or_radii, lead_times=lead_times,
+            ens_factor=args.ens_factor, sum_or_fraction=args.sum_or_fraction,
+            re_mask=True).process(cube, land_only)
+
+    result_sea = NeighbourhoodProcessing(
+        'square', radius_or_radii, lead_times=lead_times,
+        ens_factor=args.ens_factor, sum_or_fraction=args.sum_or_fraction,
+        re_mask=True).process(cube, sea_only)
+
+    # TODO: This clip should be removed once the neighbourhood code is fixed.
+    # At the moment the neighbourhood code can produce values beyond extremes
+    # in the input data.
+    if not args.no_clip:
+        apply_clipping(cube, result_land)
+        apply_clipping(cube, result_sea)
+
+    if args.intermediate_filepath is not None:
+        iris.save(result_land, '{}_land.nc'.format(
+                args.intermediate_filepath.strip('.nc')))
+        iris.save(result_sea, '{}_sea.nc'.format(
+                args.intermediate_filepath.strip('.nc')))
+
+    if args.additional_mask_filepath:
+        # Collapse with the masking dimension.
+        result_land = CollapseMaskedNeighbourhoodCoordinate(
+            args.coord_for_masking, weights=weights).process(result_land)
+
+    # Recombine cubes to be a single output.
+    combined_data = np.ma.array(
+        result_land.data.filled(1) * result_sea.data.filled(1),
+        mask=(result_land.data.mask * result_sea.data.mask))
```

Comment:
Is there anything wrong with: combined_data = result_land.data.filled(0) + result_sea.data.filled(0), since the masks are mutually exclusive by construction?

Reply:
No. I was originally concerned about any other masking I might want to persist, and slightly tied myself in knots with it. I don't think the output data needs to be masked at all anymore. I will make the above change.